src/main.rs

```diff
@@ -0,0 +1,152 @@
+#![recursion_limit = "1024"]
+
+#[macro_use]
+extern crate failure;
+extern crate glob;
+#[macro_use]
+extern crate log;
+extern crate serde;
+extern crate toml;
+
+use failure::{Fallible,ResultExt};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize)]
+pub struct ConfigInput {
+    pub collecting: CollectingInput,
+    pub reporting: ReportingInput,
+}
+
+impl ConfigInput {
+    /// Read config fragments and merge them into a single config.
+    pub fn read_configs(dirs: &[&str], app_name: &str) -> Fallible<Self> {
+        use std::io::Read;
+
+        let mut fragments = Vec::new();
+        for prefix in dirs {
+            let dir = format!("{}/{}/config.d", prefix, app_name);
+            debug!("scanning configuration directory '{}'", dir);
+
+            let wildcard = format!("{}/*.toml", dir);
+            let toml_files = glob::glob(&wildcard)?;
```

Comment:
I think I've seen this code already ;)
Please do note that I quickly put this together for zincati in order to use snippets from multiple directories, but it doesn't understand same-name overrides.

Reply:
Yes - shamelessly copied :) Thanks - yes I'd like to support same-name overrides here - pushed support for this (still need to add a test, and it likely can be optimized to not store the fragments in the BTreeMap).

Reply:
I think you could store either a `PathBuf` or a `File`, and let consumers later transform those into fragments (later reduced into a single config).

Reply:
Similar logic is at https://github.com/overdrop/overdrop-sebool/blob/master/src/od_cfg.rs#L26.